aten/src/ATen/native/cuda/UnaryGeometricKernels.cu

```diff
@@ -150,15 +152,22 @@ void cosh_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+// We manually override tanh because std::tanh does not work with thrust::complex types.
 template<typename scalar_t>
 __host__ __device__ static inline scalar_t tanh_wrapper(scalar_t v) {
   return ::tanh(v);
 }
 
+template<typename T>
+__host__ __device__ static inline thrust::complex<T> tanh_wrapper(thrust::complex<T> v) {
+  return thrust::tanh(v);
+}
+
 void tanh_kernel_cuda(TensorIterator& iter) {
   AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.dtype(), "tanh_cuda", [&]() {
```

Comment:
`AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2`

Reply:
Sorry, my bad. Missed this. Should be resolved here: https://github.com/pytorch/pytorch/pull/38486/commits/a930fce4ae0637e5a178a0c124efa9ffd98670f7. Thanks!